network/server/tchannelthrift/node/service.go

```diff
@@ -240,34 +298,48 @@ func (s *service) FetchBlocksMetadata(tctx thrift.Context, req *rpc.FetchBlocksM
 		result.Elements[i] = blocksMetadata
 	}
 
+	s.metrics.fetchBlocksMetadata.Success.Inc(1)
+	s.metrics.fetchBlocksMetadata.Latency.Stop(sw)
 	return result, nil
 }
 
 func (s *service) Write(tctx thrift.Context, req *rpc.WriteRequest) error {
+	sw := s.metrics.write.Latency.Start()
 	ctx := tchannelthrift.Context(tctx)
 
 	if req.IdDatapoint == nil || req.IdDatapoint.Datapoint == nil {
+		s.metrics.write.Error.Inc(1)
+		s.metrics.write.Latency.Stop(sw)
```

Comment:
Sure, doing it this way because that's how it was done inside database.go and namespace.go.

I'll add an ErrorLatency and fix the existing ones in db and namespace as well.